### PR TITLE
Implement Standard and Circular Queue (C++)

### LINKED ISSUE #92: Add Queue implementation in C++

### PR DESCRIPTION
**Pull request by IonesioJunior**
## Standard and Circular queue implementation

Add standard and circular queue.

## Related Issue

Fixes Issues:
fixes #92

## Types of changes

- [x] New Algorithm (non-breaking change which adds functionality)

## Checklist:
Go over all the following points, and put an `x` in all the boxes that apply.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

- [x] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	public:
+		CircularQueue(int size);
+		bool isEmpty() override;
+		bool isFull() override;
+		void enqueue(T element) override;
+		T* dequeue() override;
+		T* head() override;
+};
+
+//Queue.cpp
```
Comment (GnikDroy) [inline Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp]: Separate this to a separate CircularQueue.cpp file

File: Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	}
+	this->elements++;
+	if(this->head_index == -1 && this->tail == -1){
+		this->head_index = (this->head_index + 1 ) % this->size;
+	}
+	this->tail = (this->tail + 1 ) % this->size;
+	this->head_ptr[this->tail] = element;
+}
+
+///////////////////////////////////////////
```
Comment (GnikDroy) [inline Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp]: These lines are frankly not helpful

Comment (IonesioJunior) [reply Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp]: Done :)

File: Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp (hunk 1 above)
Comment (IonesioJunior) [reply Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp]: Done

Comment (IonesioJunior) [reply Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp]: Sorry, what is folder structure format? Data Structures/Circular Queue/C++ and Data Structures/Standard Queue?

Comment (GnikDroy) [reply Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp]: Yes that works

**Pull request merged**